### ISSUE TITLE
Invalid ternary operation code generated

### ISSUE THREAD

**Issue by MaceWindu** [agileobjects/ReadableExpressions#22]
```cs
        public enum Test
        {
            One,
            Two
        }

Expression<Func<bool, bool>> expr = flag => (flag ? Test.One : Test.Two) == Test.Two;
```

Expected:
```cs
flag => (flag ? Test.One : Test.Two) == Test.Two
or 
flag => (int)(flag ? Test.One : Test.Two) == 1
```
Actual:
```cs
flag => ((int)flag ? Test.One : Test.Two) == 1
```

**Comment by SteveWilkes** [comment]
I've added and made pass the following test:

```C#
var flagParameter = Expression.Parameter(typeof(bool), "flag");
var one = Expression.Constant(Test.One);
var two = Expression.Constant(Test.Two);
var oneOrTwo = Expression.Condition(flagParameter, one, two);
var oneOrTwoEqualsTwo = Expression.Equal(oneOrTwo, two);
var testLambda = Expression.Lambda<Func<bool, bool>>(oneOrTwoEqualsTwo, flagParameter);

var translated = ToReadableString(testLambda);

translated.ShouldBe("flag => (flag ? Test.One : Test.Two) == Test.Two");
```

...but couldn't reproduce the original translation you saw - I assume the `Test.Two` on the right-hand side of the equality test was an int value somehow?

**Comment by SteveWilkes** [comment]
Fixed in v1.13, which is [on NuGet](https://www.nuget.org/packages/AgileObjects.ReadableExpressions) and in [the Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=vs-publisher-1232914.ReadableExpressionsVisualizers).

Thanks for the feedback! :)
